src/library/dlgtagfetcher.cpp

```diff
@@ -71,23 +102,71 @@ void DlgTagFetcher::slotTrackChanged(TrackId trackId) {
+                    importedMetadata.getTrackInfo().getArtist());
+        }
+        if (!importedMetadata.getTrackInfo().getTitle().isEmpty()) {
+            updatedMetadata.refTrackInfo().setTitle(
+                    importedMetadata.getTrackInfo().getTitle());
+        }
+        if (!importedMetadata.getTrackInfo().getTrackNumber().isEmpty()) {
+            updatedMetadata.refTrackInfo().setTrackNumber(
+                    importedMetadata.getTrackInfo().getTrackNumber()
+            );
+        }
+        if (!importedMetadata.getTrackInfo().getTrackTotal().isEmpty()) {
+            updatedMetadata.refTrackInfo().setTrackTotal(
+                    importedMetadata.getTrackInfo().getTrackTotal()
+            );
+        }
+        if (!importedMetadata.getTrackInfo().getYear().isEmpty()) {
+            updatedMetadata.refTrackInfo().setYear(
+                    importedMetadata.getTrackInfo().getYear());
         }
-        if (!m_data.m_results[resultIndex]->getArtist().isEmpty()) {
-            m_track->setArtist(m_data.m_results[resultIndex]->getArtist());
+        if (!importedMetadata.getAlbumInfo().getArtist().isEmpty()) {
+            updatedMetadata.refAlbumInfo().setArtist(
+                    importedMetadata.getAlbumInfo().getArtist());
         }
-        if (!m_data.m_results[resultIndex]->getTitle().isEmpty()) {
-            m_track->setTitle(m_data.m_results[resultIndex]->getTitle());
+        if (!importedMetadata.getAlbumInfo().getTitle().isEmpty()) {
+            updatedMetadata.refAlbumInfo().setTitle(
+                    importedMetadata.getAlbumInfo().getTitle());
         }
-        if (!m_data.m_results[resultIndex]->getYear().isEmpty() &&
-             m_data.m_results[resultIndex]->getYear() != "0") {
-            m_track->setYear(m_data.m_results[resultIndex]->getYear());
+#if defined(__EXTRA_METADATA__)
+        if (!importedMetadata.getTrackInfo().getMusicBrainzArtistId().isNull()) {
+            updatedMetadata.refTrackInfo().setMusicBrainzArtistId(
+                    importedMetadata.getTrackInfo().getMusicBrainzArtistId()
+            );
         }
-        if (!m_data.m_results[resultIndex]->getTrackNumber().isEmpty() &&
-             m_data.m_results[resultIndex]->getTrackNumber() != "0") {
-            m_track->setTrackNumber(m_data.m_results[resultIndex]->getTrackNumber());
+        if (!importedMetadata.getTrackInfo().getMusicBrainzRecordingId().isNull()) {
+            updatedMetadata.refTrackInfo().setMusicBrainzRecordingId(
+                    importedMetadata.getTrackInfo().getMusicBrainzRecordingId()
+            );
         }
+        if (!importedMetadata.getTrackInfo().getMusicBrainzReleaseId().isNull()) {
+            updatedMetadata.refTrackInfo().setMusicBrainzReleaseId(
+                    importedMetadata.getTrackInfo().getMusicBrainzReleaseId()
+            );
+        }
+        if (!importedMetadata.getAlbumInfo().getMusicBrainzArtistId().isNull()) {
```

Comment:
What about the recording id and the title? I will not make up a custom correlation logic without any foundation.

I can only offer to implement one of the following options:
- Overwrite all fields unconditionally
- Set single fields one-by-one conditionally (current version)

If someone knows how to correlate the fields in a consistent manner, please do so. I refuse to do it, because I am not able to come up with a solution that is supposed to be "correct".